### ISSUE TITLE
Ability to change polygon opacity and/or color

### ISSUE THREAD

**Issue by cgmayers** [JeroenDeDauw/Maps#640]
Hello, Maps is really useful for making an interactive world map on my D&D campaign wiki! However, I'm using polygons (made in the visual editor) to define the different countries, and since they are all the same bright blue it distracts a good bit from the map underneath. Would it be possible to implement GeoJSON styles? The best case would be to edit opacity (I'd like to reduce the fillOpacity to 0 so, by clicking the invisible polygon, it'd seem like you were clicking the country on the map). Failing that, it would be nice to at least be able to change each polygon's color so they aren't the same. Would this be possible to implement?

Documentation of GeoJSON styles with Leaflet: [https://leafletjs.com/examples/geojson/](url)
Stackoverflow discussion on "fillOpacity" vs "opacity": [https://stackoverflow.com/questions/16182595/how-can-i-set-the-opacity-of-a-geojson-object-or-layer-in-leaflet](url)

**Comment by JeroenDeDauw** [comment]
See https://maps.extension.wiki/wiki/Leaflet_GeoJSON#Limitations

 I also covered it in https://professional.wiki/en/news/visual-map-editor-for-mediawiki

**Comment by cgmayers** [comment]
Thank you! This definitely sent me in the right direction. I had tried to add styles to the GeoJSON manually, but I was using full markup rather than simplestyle.

For anyone who finds this thread through Google for example, check out this page for details on simplestyle. Copy the properties exactly like this into the GeoJSON page associated with the map, and it works perfectly! [https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0#simplestyle-spec](https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0#simplestyle-spec)

JeroenDeDauw, may I suggest adding the links you posted here under the table on this page? This is the page I studied most in trying to make this work, and other users might do the same. [https://maps.extension.wiki/wiki/Customizing_Leaflet_maps](https://maps.extension.wiki/wiki/Customizing_Leaflet_maps)

Again thank you very much for your response.
